### PR TITLE
refactor: PRSDM-5464 setup semantic release

### PR DESCRIPTION
**Pull request by Zalaras**
Add semantic release and replaced TravisCI with Github Actions

### Description
- Add semantic release
- Replace TravisCI with Github Actions
- Remove build scripts
- Update Node version for TravisCI
- Update Readme
- Update PR template
- Update github token and rename to `GITHUB_TOKEN` from `GITHUB_KEY` (required for semantic release)

[Github Action dry run](https://github.com/SPANDigital/presidium-js/actions/runs/9973864082/job/27560149522?pr=127)

Dry Run:
```
[11:40:30 PM] [semantic-release] › ⚠  Skip v2.13.2-rfv.1 tag creation in dry-run mode
[11:40:30 PM] [semantic-release] › ⚠  Skip step "publish" of plugin "@semantic-release/exec" in dry-run mode
[11:40:30 PM] [semantic-release] › ⚠  Skip step "publish" of plugin "@semantic-release/github" in dry-run mode
[11:40:30 PM] [semantic-release] › ⚠  Skip step "success" of plugin "@semantic-release/exec" in dry-run mode
[11:40:30 PM] [semantic-release] › ⚠  Skip step "success" of plugin "@semantic-release/github" in dry-run mode
[11:40:30 PM] [semantic-release] › ✔  Published release 2.13.2-rfv.1 on develop channel
[11:40:30 PM] [semantic-release] › ℹ  Release note for version 2.13.2-rfv.1:
## 2.13.2-rfv.1 (2024-07-15)

### Bug Fixes

    * add semantic release for builds and test trigger (0d78c16)
```

### Issue
[PRSDM-5464](https://spandigital.atlassian.net/browse/PRSDM-5464)

### Checklist before merging

* [ ] Is this code covered by tests?
* [ ] Is the documentation updated for this change?

### CODE REVIEW

**Review by CharlRitterDev — APPROVED** [review]
And I thought my latest FE PRs were big 😝

**Comment by Zalaras** [comment]
@CharlRitterDev lol it's the lock file...

To everyone else: I'm working on the build fail. Issue with Node version and the Ubuntu version that TravisCI uses

**Review by CharlRitterDev — APPROVED** [review]

**Review by meyerhp — COMMENTED** [review]
Looks good overall. Posted a few comments below

File: release.config.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          hidden: false,
+        },
+      ],
+    },
+  }],
+  [
+    "@semantic-release/exec",
+    {
+      prepareCmd: "zip -r -qq ${nextRelease.version} dist && printf '${nextRelease.version}' > VERSION",
+      successCmd: "TRAVIS_TAG=${nextRelease.gitTag}",
```
[the remark this answers is not in the document]
Comment (Zalaras) [reply release.config.js:31]: This is what happens when you change CI and miss some changes... Thanks

File: release.config.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      changelogFile: "CHANGELOG.md",
+    },
+  ],
+  ['@semantic-release/git',
+    {
+      assets: ['CHANGELOG.md', 'VERSION'],
+    },
+  ])
+} else {
+  ['@semantic-release/git',
```
[the remark this answers is not in the document]
Comment (Zalaras) [reply release.config.js:55]: Thanks, going to update presidium-services too...

File: release.config.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          section: "Refactors",
+          hidden: false,
+        },
+      ],
+    },
+  }],
+  [
+    "@semantic-release/exec",
+    {
+      prepareCmd: "zip -r -qq ${nextRelease.version} dist && printf '${nextRelease.version}' > VERSION",
```
[the remark this answers is not in the document]
Comment (Zalaras) [reply release.config.js:30]: again another Travis change I missed, thanks!

**Review by meyerhp — APPROVED** [review]

**Pull request merged**